apps/data_handler/db/crud.py

```diff
@@ -942,15 +997,41 @@ def create_debt_transfer_event(
         finally:
             db.close()
 
-    def create_bearing_collateral_burn_event(
+    def create_interest_rate_model_event(
         self, protocol_id: str, event_name: str, block_number: int, event_data: dict
     ) -> None:
         """
-        Creates a BearingCollateralBurnEventModel record in the database.
+        Creates a InterestRateModelEventModel record in the database.
         """
         db = self.Session()
         try:
-            event = BearingCollateralBurnEventModel(
+            event = InterestRateModelEventModel(
+                protocol_id=protocol_id,
+                event_name=event_name,
+                block_number=block_number,
+                debt_token=event_data.get("debt_token"),
+                lending_index=event_data.get("lending_index"),
+                borrow_index=event_data.get("borrow_index"),
+            )
+            db.add(event)
+            db.commit()
+            logger.info(f"InterestRateModel event saved: {event}")
+        except SQLAlchemyError as e:
+            db.rollback()
+            logger.error(f"Error creating InterestRateModelEventModel: {e}")
+            raise e
+        finally:
+            db.close()
+
+    def create_non_interest_bearing_collateral_burn_event(
+        self, protocol_id: str, event_name: str, block_number: int, event_data: dict
+    ) -> None:
+        """
+        Creates a NonInterestBearingCollateralBurnEventModel record in the database.
+        """
+        db = self.Session()
```

Comment:
why not use write_to_db method?
please change it in another places as well

Reply:
I refactored the code to use the `write_to_db` method and added a new method, `get_all_events_from_models`, to the base `DBConnector` class